wallet/src/Pos/Wallet/Web/Methods/Logic.hs

```diff
@@ -96,59 +93,60 @@ getWAddress cachedAccModifier cAddr = do
         RequestError $ sformat ("No account with id "%build%" found") accId
-    gatherAddresses modifier dbAddrs = do
-        let memAddrs = sortedInsertions modifier
+    gatherAddresses addrModifier dbAddrs = do
+        let memAddrs = sortedInsertions addrModifier
             relatedMemAddrs = filter ((== accId) . addrMetaToAccount) memAddrs
             -- @|relatedMemAddrs|@ is O(1) while @dbAddrs@ is large
             unknownMemAddrs = filter (`notElem` dbAddrs) relatedMemAddrs
         dbAddrs <> unknownMemAddrs
 
-getWalletIncludeUnready :: MonadWalletWebMode m => Bool -> CId Wal -> m CWallet
-getWalletIncludeUnready includeUnready cAddr = do
-    meta       <- getWalletMetaIncludeUnready includeUnready cAddr >>= maybeThrow noWSet
-    wallets    <- getAccountsIncludeUnready includeUnready (Just cAddr)
-    let walletsNum = length wallets
-    balance    <- mkCCoin . unsafeIntegerToCoin . sumCoins <$>
-                     mapM (decodeCTypeOrFail . caAmount) wallets
-    hasPass    <- isNothing . checkPassMatches emptyPassphrase <$> getSKById cAddr
-    passLU     <- getWalletPassLU cAddr >>= maybeThrow noWSet
-    pure $ CWallet cAddr meta walletsNum balance hasPass passLU
-  where
-    noWSet = RequestError $
-        sformat ("No wallet with address "%build%" found") cAddr
-
-getWallet :: MonadWalletWebMode m => CId Wal -> m CWallet
-getWallet = getWalletIncludeUnready False
-
 getAccountsIncludeUnready
     :: MonadWalletWebMode m
     => Bool -> Maybe (CId Wal) -> m [CAccount]
 getAccountsIncludeUnready includeUnready mCAddr = do
-    whenJust mCAddr $ \cAddr -> getWalletMetaIncludeUnready includeUnready cAddr `whenNothingM_` noWSet cAddr
+    whenJust mCAddr $ \cAddr -> getWalletMetaIncludeUnready includeUnready cAddr `whenNothingM_` noWallet cAddr
     accIds <- maybe getAccountIds getWalletAccountIds mCAddr
     let groupedAccIds = fmap reverse $ HM.fromListWith mappend $
                         accIds <&> \acc -> (aiWId acc, [acc])
     concatForM (HM.toList groupedAccIds) $ \(wid, walAccIds) ->
          fixCachedAccModifierFor wid $ forM walAccIds . getAccount
   where
-    noWSet cAddr = throwM . RequestError $
+    noWallet cAddr = throwM . RequestError $
+        -- TODO No WALLET with id ...
+        -- dunno whether I can fix and not break compatible w/ daedalus
         sformat ("No account with id "%build%" found") cAddr
 
 getAccounts
     :: MonadWalletWebMode m
     => Maybe (CId Wal) -> m [CAccount]
 getAccounts = getAccountsIncludeUnready False
 
+getWalletIncludeUnready :: MonadWalletWebMode m => Bool -> CId Wal -> m CWallet
+getWalletIncludeUnready includeUnready cAddr = do
+    meta       <- getWalletMetaIncludeUnready includeUnready cAddr >>= maybeThrow noWallet
+    accounts   <- getAccountsIncludeUnready includeUnready (Just cAddr)
+    let accountsNum = length accounts
+    balance    <- mkCCoin . unsafeIntegerToCoin . sumCoins <$>
+                     mapM (decodeCTypeOrFail . caAmount) accounts
```

Comment:
I'm wondering if we could generalise this pattern (which I have seen it occurred already 3 times during this PR) to yield slightly better memory allocation by not collecting all the accounts/addresses at once and then summing them up, but rather doing a monadic fold with a strict accumulator so that we would sum the coins whilst we traverse the accounts/addresses. What do you think?